### ISSUE TITLE
Sarasa Gothic font supported on macOS but not on Linux

### ISSUE THREAD

**Issue by terrywang** [kovidgoyal/kitty#3021]
**Describe the bug**

[Sarasa Gothic](https://github.com/be5invis/Sarasa-Gothic) is a CJK programming font based on Iosevka and Source Han Sans.

I've started using Sarasa Gothic with kitty on macOS for months (a nerd friend showed me and I love the way it looks), very happy with it.

When trying to do the same for kitty on Linux (kitty's my secondary terminal emulator in addition to `Konsole`, that's why I just discovered the issue), it fell back to `DroidSansMono`, `kitty list-fonts` did not show the Sarasa Gothic fonts at all (tried both specific ttf and ttc, force rebuild font cache, etc.).

```bash
kitty --debug-font-fallback
[285 23:24:36.425212] Preloaded font faces:
[285 23:24:36.425284] normal face: /usr/share/fonts/droid/DroidSansMono.ttf:0
[285 23:24:36.425317] bold face: /usr/share/fonts/droid/DroidSansMono.ttf:0
[285 23:24:36.425344] italic face: /usr/share/fonts/droid/DroidSansMono.ttf:0
[285 23:24:36.425369] bi face: /usr/share/fonts/droid/DroidSansMono.ttf:0
```

A big of digging revealed that kitty does NOT support bitmap fonts
source 1: issue #97
source 2: [kitty-cat](https://github.com/adi1090x/kitty-cat)
source 3: [`kitty-bitmap` @ AUR](https://aur.archlinux.org/packages/kitty-bitmap)

However, kitty supports `Iosevka` well on Linux, as a derivative Sarasa Gothic I would assume it would be supported but I was wrong.

### The Question

Could anyone please confirm if Sarasa Gothic Term fonts are bitmap fonts?

If so, why does Sarasa Gothic work with kitty on macOS but not on Linux?

```bash
$  kitty --version
kitty 0.19.1 created by Kovid Goyal
$  kitty --debug-font-fallback
[285 23:32:57.485745] Preloaded font faces:
[285 23:32:57.485804] normal face: /Users/terry.wang/Library/Fonts/sarasa-regular.ttc
[285 23:32:57.485818] bold face: /Users/terry.wang/Library/Fonts/sarasa-semibold.ttc
[285 23:32:57.485833] italic face: /Users/terry.wang/Library/Fonts/sarasa-italic.ttc
[285 23:32:57.485847] bi face: /Users/terry.wang/Library/Fonts/sarasa-semibolditalic.ttc
```

**To Reproduce**
1. Install Sarasa Gothic font (ttc via package manager, or specifc ttf in `~/.fonts` and/or `~/.local/share/fonts`)
2. Rebuild font cache (`fc-cache -fv  /path/to/font`)
3. List font `kitty list-fonts`
4. Sarasa Gothic Term * are not shown

**Expected behavior**
Expect kitty to support Sarasa Gothic as it does on macOS.

**Screenshots**
NA

**Environment details**
OS: Manjaro (KDE Plasma / LXQt)

```bash
$ kitty --debug-config
kitty 0.19.1 created by Kovid Goyal
Linux macbook 5.8.11-1-MANJARO #1 SMP PREEMPT Wed Sep 23 14:35:40 UTC 2020 x86_64
Manjaro Linux \r  (\n) (\l)
DISTRIB_ID=ManjaroLinux
DISTRIB_RELEASE=20.1.1
DISTRIB_CODENAME=Mikah
DISTRIB_DESCRIPTION="Manjaro Linux"
Loaded config files: /home/terry/.config/kitty/kitty.conf
Running under: X11

Config options different from defaults:
background           Color(red=44, green=44, blue=44)
color1               Color(red=241, green=119, blue=121)
color10              Color(red=153, green=204, blue=153)
color11              Color(red=255, green=204, blue=102)
color12              Color(red=102, green=153, blue=204)
color13              Color(red=204, green=153, blue=204)
color14              Color(red=102, green=204, blue=204)
color15              Color(red=255, green=254, blue=254)
color2               Color(red=153, green=204, blue=153)
color3               Color(red=255, green=204, blue=102)
color4               Color(red=102, green=153, blue=204)
color5               Color(red=204, green=153, blue=204)
color6               Color(red=102, green=204, blue=204)
color7               Color(red=255, green=254, blue=254)
color8               Color(red=0, green=0, blue=0)
color9               Color(red=241, green=119, blue=121)
font_family          Sarasa Term SC Regular
foreground           Color(red=204, green=204, blue=204)
selection_background Color(red=80, green=80, blue=80)
selection_foreground Color(red=44, green=44, blue=44)
```

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/faq.html#kitty-is-not-able-to-use-my-favorite-font

**Comment by terrywang** [comment]
@kovidgoyal thanks for the quick response.

It works on macOS and with Konsole, I blindly assumed that it should work. Somehow GoogleFu didn't lead me to FAQ.

I'll try the workaround in FAQ (there is one mentioned in #97 as well). If it works I'll create a blog post to help other kitty users who run into the same issue.

**Comment by terrywang** [comment]
## Update

`fontconfig` config file override trick did not work.

TL;DR
Caused by bug (problematic glyphs) in Sarasa Gothic v0.14.3 and earlier, which subsequently fails to please fontconfig on Linux where it failed to calculate the spacing property based on actual glyph widths in the font.

More info, see https://github.com/be5invis/Sarasa-Gothic/issues/192

**Comment by terrywang** [comment]
Sarasa Gothic upstream has fixed the problem, use `v0.14.5` or later to satisfy Linux `fontconfig`, it can correctly calculate the spacing property based on actual glyph widths in the font ;-)

```bash
fc-list : family spacing outline scalable | awk '/spacing=100/ || /spacing=90/' | awk '/outline=True/ && /scalable=True/' | grep -i Sarasa
Sarasa Term SC:spacing=90:outline=True:scalable=True
```

https://github.com/be5invis/Sarasa-Gothic/issues/192
